.ci/Jenkinsfile

```
@@ -259,6 +264,66 @@ pipeline {
                gitHubCheckSha1: env.GIT_BASE_COMMIT)
       }
     }
+    stage('extended windows') {
+      when {
+        // Always when running builds on branches/tags
+        // Enable if extended windows support related changes.
+        beforeAgent true
+        anyOf {
+          not { changeRequest() }
+          expression { return isExtendedWindowsEnabled() && env.ONLY_DOCS == "false"}
+        }
+      }
+      failFast false
+      matrix {
+        agent {label "${PLATFORM} && windows-immutable"}
+        options { skipDefaultCheckout() }
+        axes {
+          axis {
+            name 'PLATFORM'
+            values 'windows-8', 'windows-10', 'windows-11', 'windows-2012-r2'
```

Is this list the one we wanna test with?

@ph @cmacknz and @jlind23

---

This is for the PR platform check, we should probably have a newer windows than 2012-r2.

@cmacknz What do you think about windows-8 our oldest platform, windows-11 and Windows server 2022?

---

`Windows-2012-r2` is just the server version of `windows-8` so you probably don't need to test both.

I am definitely in favour of adding Windows server 2022 to test the newest version. Windows 10 isn't EOL until 2025 so it will be around for a while still, it might be worth keeping it in the list.

---

sound good to me.

---

windows-2022 is already in the default stage, this particular new stage is an extended one so it runs in some specific cases defined in `isExtendedWindowsEnabled` 
```suggestion
            values 'windows-8', 'windows-10', 'windows-11'
```